### PR TITLE
MODINREACH-337  Update the module to Spring boot v3.0.2

### PR DESCRIPTION
**Pull request by tatsiana-tarhonskaya**
## Purpose
Update the module to Spring boot v3.0.2

## Approach
Module updated to use java 17 https://issues.folio.org/browse/MODINREACH-336 and spring boot updated to v3 https://issues.folio.org/browse/MODINREACH-337

## Learning
1) https://wiki.folio.org/display/TC/JDK+17+and+Java+17
2) https://wiki.folio.org/pages/viewpage.action?spaceKey=~Taras_Spashchenko&title=Migration+to+Spring+Boot+v3.0.0+How+to
3) QueryHints#HINT_PASS_DISTINCT_THROUGH not longer needed: https://github.com/hibernate/hibernate-orm/blob/6.0/migration-guide.adoc#distinct
4) replacing of WebSecurityConfigurerAdapter: https://spring.io/blog/2022/02/21/spring-security-without-the-websecurityconfigureradapter
5) update openAPI generator version from 4 to 6 caused some changes:
   - another names generated. To decrease changes, following properties used: inlineSchemaNameDefaults: arrayItemSuffix= and some updates to schema itself.
   - for  arrays with "uniqueItems": true Set is generated instead of List
   - usage of "allOf" in json schema does not automatically produce inherited classes now. There should be discriminator field used to achieve this : 
   https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#fixed-fields-20
   https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#composition-and-inheritance-polymorphism
   Tried to do this way first, but such implementation would require all the responses of such type to be updated with this new discriminator field. However this inheritance is used in just one class for handling the responses: https://github.com/folio-org/mod-inn-reach/compare/master...MODINREACH-336#diff-52b6a39b6787d870838b307c053a6e7064abb64c74259d67667c047268c6346dR101
So decided to use another approach instead, to create an interface https://github.com/folio-org/mod-inn-reach/compare/master...MODINREACH-336#diff-2f34db20050ba6cf0c6b16e4e13aea20d9fd6d80e2c550517222821a67e41ddcR1 and use it along with adding "x-implements" : https://openapi-generator.tech/docs/generators/spring/

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -108,16 +114,6 @@
         </exclusion>
       </exclusions>
     </dependency>
-    <dependency>
```
Comment (tatsiana-tarhonskaya) [inline pom.xml:111]: spring batch not used anymore, can be removed

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -45,16 +43,17 @@ spring:
       key-store-location: ${KAFKA_SSL_KEYSTORE_LOCATION:}
       trust-store-password: ${KAFKA_SSL_TRUSTSTORE_PASSWORD:}
       trust-store-location: ${KAFKA_SSL_TRUSTSTORE_LOCATION:}
-  batch:
-    job:
-      enabled: false
-    initializer:
```
Comment (tatsiana-tarhonskaya) [inline src/main/resources/application.yml:51]: spring batch properties not needed

File: src/main/resources/swagger.api/schemas/d2ir/bibInfoResponseDTO.json [hunk 3]
```diff
@@ -7,6 +7,7 @@
   ],
   "description": "D2IR Bib info response",
   "type": "object",
+  "x-implements": "org.folio.innreach.domain.dto.InnReachResponseData",
```
Comment (tatsiana-tarhonskaya) [inline src/main/resources/swagger.api/schemas/d2ir/bibInfoResponseDTO.json:10]: the reason for this is described in the PR description

File: src/main/resources/swagger.api/api-common.yaml [hunk 4]
```diff
@@ -4,7 +4,7 @@ components:
       type: string
       format: uuid
     error:
-      $ref: schemas/error.json
+      $ref: schemas/error.json#/Error
```
Comment (tatsiana-tarhonskaya) [inline src/main/resources/swagger.api/api-common.yaml:7]: This change is to leave generated pojo as is (Error). Without this change there's invalid pojo generated for some reason..

**Review by TarasSpashchenko — APPROVED** [review]

**Review by arin-epm — APPROVED** [review]

**Review by gurleenkaurbp — APPROVED** [review]

**Pull request merged**